packages/terra-pills/src/Pill.jsx

```diff
@@ -0,0 +1,212 @@
+  ariaControls: undefined,
+  title: undefined,
+};
+
+const Pill = (props) => {
+  const {
+    label,
+    id,
+    onRemove,
+    onSelect,
+    refCallback,
+    ariaHasPopup,
+    ariaExpanded,
+    ariaControls,
+    title,
+    intl,
+    ...customProps
+  } = props;
+
+  const pillRef = useRef();
+
+  useEffect(() => {
+    if (refCallback) {
+      refCallback(pillRef.current);
+    }
+  }, [refCallback]);
+
+  const handleOnClick = () => {
+    onSelect();
+  };
+
+  const handleOnRemove = () => {
+    onRemove();
+  };
+
+  const handleOnKeyDown = (event) => {
+    pillRef.current.setAttribute('data-terra-pills-show-focus-styles', 'true');
+    if ((event.keyCode === KEY_RETURN || event.keyCode === KEY_SPACE) && onSelect) {
+      onSelect();
+      event.preventDefault();
+    } else if ((event.keyCode === KEY_DELETE || event.keyCode === KEY_BACK_SPACE) && onRemove) {
+      onRemove();
+    }
+  };
+
+  const handleOnMouseDown = () => {
+    pillRef.current.setAttribute('data-terra-pills-show-focus-styles', 'false');
+  };
+
+  const pillProps = {};
+  if (onSelect || onRemove) {
+    pillProps.tabIndex = '0';
+    pillProps.onKeyDown = handleOnKeyDown;
+    pillProps.onMouseDown = handleOnMouseDown;
+    pillProps.role = 'button';
+  }
+
+  const pillButtonProps = {};
+  if (onSelect) {
+    pillButtonProps.title = title;
+    pillButtonProps.onClick = handleOnClick;
+  }
+
+  const removeButtonProps = {};
+  if (onRemove) {
+    removeButtonProps.onClick = handleOnRemove;
+  }
+
+  let pillInteractionHint;
+  if (onSelect && onRemove) {
+    pillInteractionHint = intl.formatMessage({ id: 'Terra.pills.pillHint.selectableAndRemovable' });
+  } else if (onSelect) {
+    pillInteractionHint = intl.formatMessage({ id: 'Terra.pills.pillHint.selectable' });
+  } else if (onRemove) {
+    pillInteractionHint = intl.formatMessage({ id: 'Terra.pills.pillHint.removable' });
+  }
+
+  const theme = React.useContext(ThemeContext);
+
+  const pillClassNames = classNames(
+    cx([
+      'pill-container',
+      { 'is-focusable': !!onSelect || !!onRemove },
+      { 'is-selectable': !!onSelect },
+      { 'is-selectable-and-removable': !!onSelect && !!onRemove },
+      theme.className,
+    ]),
+    customProps.className,
+  );
+
+  const pillLabelClassNames = cx([
+    'pill-label',
+    { 'is-selectable': !!onSelect },
+    { 'is-removable': !!onRemove },
+    { 'is-selectable-and-removable': !!onSelect && !!onRemove },
+  ]);
+
+  const removeButtonClassNames = cx([
+    'pill-remove-button',
+  ]);
+
+  return (
+    <div
+      {...customProps}
+      {...pillProps}
+      id={id}
+      ref={pillRef}
+      aria-haspopup={!!onSelect && ariaHasPopup ? true : undefined}
+      aria-expanded={!onSelect ? undefined : ariaExpanded}
+      aria-controls={!onSelect ? undefined : ariaControls}
+      data-terra-pills-show-focus-styles
+      data-terra-pill
+      className={pillClassNames}
+    >
+      <div
+        {...pillButtonProps}
```

Comment:
title prop which is required for tooltip will be added only when onSelect is provided 
line number 133. So this makes tooltip to be not displayed for non-selectable pill with long text. 
```js
 const pillButtonProps = {};
  if (onSelect) {
    pillButtonProps.title = title;
    pillButtonProps.onClick = handleOnClick;
  }
```
So I think title should be added to [parent div](https://github.com/cerner/terra-core/blob/fb4c4c75f9b198a1180dc942d4e5d142e7ca5d89/packages/terra-pills/src/Pill.jsx#L178) of pill instead  of adding it to button div.